### PR TITLE
Added geolocation + search day

### PR DESCRIPTION
**Pull request by yaliza**
Now you can add your current geolocation in edit mode and you can search a day(s) by title.

### CODE REVIEW

File: app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayEditModeFragment.java [hunk 1]
```diff
@@ -242,4 +271,22 @@ private void closeFABMenu() {
...
+        String locationProvider = LocationManager.NETWORK_PROVIDER;
+        location = locationManager.getLastKnownLocation(locationProvider);
+        Button button = new Button(getActivity());
+        button.setText(R.string.show_location);
+        button.setOnClickListener(v->{
+            LocationMapManager.showLocation(getActivity(),
+                String.valueOf(location.getLatitude()), String.valueOf(location.getLongitude()));
+        });
+
+        fabAddLocation.setClickable(false);
```
Comment (Kalyanova) [inline app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayEditModeFragment.java]: Maybe to move it (disabling `fabAddLocation `button) up early, before `addLocation()` method invocation?

File: app/src/main/java/by/paranoidandroid/dailyvisualizer/view/utils/LocationMapManager.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import android.content.Intent;
+import android.net.Uri;
+import android.widget.Toast;
+import by.paranoidandroid.dailyvisualizer.R;
+
+public class LocationMapManager {
+    public static void showLocation(Context context, String lattitude, String longtitude){
+      Uri gmmIntentUri = Uri.parse("geo:" + lattitude + "," + longtitude);
+      Intent mapIntent = new Intent(Intent.ACTION_VIEW, gmmIntentUri);
+      mapIntent.setPackage("com.google.android.apps.maps");
```
Comment (Kalyanova) [inline app/src/main/java/by/paranoidandroid/dailyvisualizer/view/utils/LocationMapManager.java]: Maybe is it better to move "com.google.android.apps.maps" to Constants interface?

File: app/src/main/java/by/paranoidandroid/dailyvisualizer/view/utils/LocationMapManager.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      if (mapIntent.resolveActivity(context.getPackageManager()) != null) {
+        context.startActivity(mapIntent);
+      } else {
+        Toast.makeText(context, R.string.install_maps, Toast.LENGTH_SHORT);
```
Comment (Kalyanova) [inline app/src/main/java/by/paranoidandroid/dailyvisualizer/view/utils/LocationMapManager.java]: Maybe to add here `show()` invocation? As:
`Toast.makeText(context, R.string.install_maps, Toast.LENGTH_SHORT).show();`

File: app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayFragment.java [hunk 4]
```diff
@@ -104,9 +103,18 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
                 } else {
                     ivDay.setImageDrawable(null);
                 }
+                if(day.getLatitude() != null){
+                    btShowLocation.setVisibility(View.VISIBLE);
+                    btShowLocation.setOnClickListener(v->{
+                      LocationMapManager.showLocation(getActivity(), day.getLatitude(), day.getLongitude());
+                    });
+                } else {
+                    btShowLocation.setVisibility(View.INVISIBLE);
```
Comment (Kalyanova) [inline app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayFragment.java]: Maybe is it better to change visibility from `View.INVISIBLE` to `View.GONE`?

File: app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayFragment.java [hunk 5]
```diff
@@ -104,9 +103,18 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
...
+                }
             } else {
                 tvDescription.setText(getString(R.string.label_empty_day));
                 ivDay.setImageDrawable(null);
+                btShowLocation.setVisibility(View.INVISIBLE);
```
Comment (Kalyanova) [inline app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayFragment.java]: And here too (maybe is it better to change visibility from `View.INVISIBLE` to `View.GONE`?)

File: app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/SearchFragment.java [hunk 6]
```diff
@@ -18,6 +32,43 @@ public void onCreate(Bundle savedInstanceState) {
...
+        progressBar = view.findViewById(R.id.pb_search);
+
+        searchView = view.findViewById(R.id.search_view);
+        searchView.setOnQueryTextListener(new OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String s) {
+                progressBar.setVisibility(View.VISIBLE);
+                recyclerView.setVisibility(View.GONE);
+                tvNothingToShow.setVisibility(View.GONE);
+                model.setSearchQuery(s);
```
Comment (Kalyanova) [inline app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/SearchFragment.java]: I've added a new commit to this branch:
"A little fix: trim() method added (to remove any leading and trailing whitespaces)"
https://github.com/Kalyanova/DailyVisualizer/pull/18/commits/e6cd269fe62e8c607ddc4e5274b2bcd02d641842

Comment (Kalyanova) [reply app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/SearchFragment.java]: And one more commit to this branch :) To clear SearchView by clicking on close button.
"Added OnCloseListener on SearchView"
https://github.com/Kalyanova/DailyVisualizer/pull/18/commits/db454bafc95b39983951ad24d8c77f98667f8102

File: app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayEditModeFragment.java [hunk 7]
```diff
@@ -242,4 +271,22 @@ private void closeFABMenu() {
...
         fabAddLocation.animate().translationY(0);
     }
+
+    //permission already checked
+    @SuppressLint("MissingPermission")
+    private void addLocation() {
+        LocationManager locationManager = (LocationManager) getActivity()
+            .getSystemService(Context.LOCATION_SERVICE);
+        String locationProvider = LocationManager.NETWORK_PROVIDER;
+        location = locationManager.getLastKnownLocation(locationProvider);
```
Comment (Kalyanova) [inline app/src/main/java/by/paranoidandroid/dailyvisualizer/view/fragments/DayEditModeFragment.java]: And maybe to wrap it in try-catch?

**Review by Kalyanova — APPROVED** [review]
Great job! 👍

**Review by ErgerSusha — APPROVED** [review]

**Review by filak8 — APPROVED** [review]

**Pull request merged**